From d0e1489cfd77093da03bff080b9a9a93ec8a2728 Mon Sep 17 00:00:00 2001
From: Evan Quan <evan.quan@amd.com>
Date: Thu, 13 Apr 2017 14:43:58 +0800
Subject: [PATCH 2017/2094] drm/amdgpu: remove the unused vm_manager lock - the
 vm_manager lock was removed by "drm/amdgpu: split VMID management by VMHUB" -
 fine granularity lock was introduced stead

Change-Id: I5bfb4e1d1ad41b30f0bd2f1d6a4cb412eadf127b
Signed-off-by: Evan Quan <evan.quan@amd.com>
Reviewed-by: Junwei Zhang <Jerry.Zhang@amd.com>
Reviewed-by: Yong Zhao <Yong.Zhao@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
index 1a43253..a693e96 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
@@ -2352,8 +2352,6 @@ int amdgpu_vm_init(struct amdgpu_device *adev, struct amdgpu_vm *vm,
 
 	vm->is_kfd_vm = is_kfd_vm;
 	if (is_kfd_vm) {
-		mutex_lock(&adev->vm_manager.lock);
-
 		if (adev->vm_manager.n_kfd_vms++ == 0) {
 			/* First KFD VM: enable compute power profile */
 			if (adev->pp_enabled)
@@ -2363,7 +2361,6 @@ int amdgpu_vm_init(struct amdgpu_device *adev, struct amdgpu_vm *vm,
 				adev->pm.funcs->switch_power_profile(adev,
 						AMD_PP_COMPUTE_PROFILE);
 		}
-		mutex_unlock(&adev->vm_manager.lock);
 	}
 
 	return 0;
@@ -2418,8 +2415,6 @@ void amdgpu_vm_fini(struct amdgpu_device *adev, struct amdgpu_vm *vm)
 	int i;
 
 	if (vm->is_kfd_vm) {
-		mutex_lock(&adev->vm_manager.lock);
-
 		WARN(adev->vm_manager.n_kfd_vms == 0, "Unbalanced number of KFD VMs");
 
 		if (--adev->vm_manager.n_kfd_vms == 0) {
@@ -2431,7 +2426,6 @@ void amdgpu_vm_fini(struct amdgpu_device *adev, struct amdgpu_vm *vm)
 				adev->pm.funcs->switch_power_profile(adev,
 						AMD_PP_GFX_PROFILE);
 		}
-		mutex_unlock(&adev->vm_manager.lock);
 	}
 
 	amd_sched_entity_fini(vm->entity.sched, &vm->entity);
-- 
2.7.4

